roles/mirror_ocp_release/tasks/artifacts.yml

```diff
@@ -11,20 +11,26 @@
   when:
     - mor_force or not _mor_target.stat.exists
   block:
-    - name: "Extract installer and metadata from release image"
-      ansible.builtin.shell: >
-        flock -x {{ mor_cache_dir }}/{{ mor_version }}/release_extract.lock -c '
```

Comment:
With this new approach, use of filesystem locks is not needed anymore.

Reply:
Have you considered scenarios where two jobs run concurrently, both extracting to a temporary location and writing to `mor_cache_dir`? How do you prevent conflicts in such cases? Implementing a mechanism like a lock might help avoid these issues.

Reply:
that's a fair point. I was accepting the risk of facing such scenarios provided moving files around the filesystem is must faster than running the operations directly on the mor_cache_dir.

But at this point it's right what we could just limit the protected zone to the task were we copy the files to the cache directory once they have been processed in the temporary directory.

Reply:
here comes another thought. The way this implementation works, the problematic task would be when we run the copy module to move the files from the temporary directory to the cache directory. Alternatively or in combination of the lock usage, we can add the parameter "force: false" to the module call, so ansible won't replace a file that already exists, even if the contents are different.
The question here is whether it'd be safe to assume the artifact won't change between jobs deploying the same OCP release.
My guess is the files won't change if the jobs are running concurrently, but would those artifacts change between jobs running with, say, days of difference?